### DISCUSSION TITLE
No proper data to initialize the Model of component 'tok2vec'

### DISCUSSION THREAD

**Discussion by lorenzo82** [explosion/spaCy#11428] [Help: Coding & Implementations] [answered]
I am encountering the following error when running the spacy train.

_**"AssertionError: [E923] It looks like there is no proper sample data to initialize the Model of component 'tok2vec'. To check your input data paths and annotation, run: python -m spacy debug data config.cfg and include the same
config override values you would specify for the 'spacy train' command."**_

I have created two json file (train.json and dev.json) and I converted them into binary .spacy files.
I have download the base_config.cfg and updated it into a config.cfg file.
After the execution of "python -m spacy debug data config.cfg" I obtained the folowing error:
_**"AssertionError: [E923] It looks like there is no proper sample data to initialize the Model of component 'tok2vec'. To check your input data paths and annotation, run: python -m spacy debug data config.cfg and include the same
config override values you would specify for the 'spacy train' command."**_

I do not understand what am I doing wrong.

Thank you.

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @lorenzo82 ,

My guess is that something wrong happened during the conversion of the JSON file to the spaCy file. I wonder where the JSON file came from? (is this from the v2.x version of spaCy?). Assuming you converted it using [convert](https://spacy.io/api/cli#convert), perhaps the next step is to manually inspect the spaCy files and check for empty docs.

```python
import spacy
from spacy.tokens import DocBin

nlp = spacy.blank("xx")  # or a language code, e.g. `en`
doc_bin = DocBin().from_disk("path/to/file.spacy")
docs = list(doc_bin.get_docs(nlp.vocab))
```

There could be several reasons as to why this happens. Perhaps it's in the formatting of the JSON file or an error during the conversion process.

**Comment by lorenzo82** [comment]
In order to give you more element, you can find the formatting of the JSON file.

`[
    [
        "Global analysis and simulation of land-use change associated with urbanization.",
        {
            "entities": [
                [
                    34,
                    42,
                    "LC"
                ]
            ]
        }
    ],
    [
        "Exploring subpixel learning algorithms for estimating global land cover frac- tions from satellite data using high performance computing.",
        {
            "entities": [
                [
                    61,
                    71,
                    "LC"
                ]
            ]
        }
    ]
]`

**Reply by ljvmiranda921** [reply]
Hi @lorenzo82 ,

The `convert` CLI does not support this format. It's from an inline demo example and not the JSON format back in v2. It's better if you do the conversion programmatically. I suggest taking a look at this [NER demo project](https://github.com/explosion/projects/tree/v3/pipelines/ner_demo) and check the [conversion script](https://github.com/explosion/projects/blob/v3/pipelines/ner_demo/scripts/convert.py) as a starting point.

**Reply by lorenzo82** [reply]
Thank you,
I solved the problem.

**Reply by achal648** [reply]
Hi @lorenzo82 ,

Could you please share how you solved it? I did the above check mentioned by @ljvmiranda921 and cross-checked my conversion script; nothing seems wrong but still getting this error.

**Reply by ljvmiranda921** [reply]
Hi @achal648 , can you give an example of your dataset before passing it to the conversion script?

**Reply by achal648** [reply]
Hi @ljvmiranda921 The issue was with the training and dev data path that I was passing through CLI command. It is solved now, thanks!

**Reply by sunilksamanta** [reply]
@ljvmiranda921 
Hi this is my format. `[
    [
        "Requesting Hiring for the Radar System featuring Viking Line equipment",
        {
            "entities": [
                [
                    26,
                    38,
                    "EQUIPMENT"
                ],
                [
                    49,
                    60,
                    "BRAND"
                ],
                [
                    11,
                    17,
                    "SERVICE"
                ]
            ]
        }
    ],
    [
        "Desire Repair for the Refrigeration Units utilizing Hamburg equipment",
        {
            "entities": [
                [
                    22,
                    41,
                    "EQUIPMENT"
                ],
                [
                    52,
                    63,
                    "BRAND"
                ],
                [
                    7,
                    13,
                    "SERVICE"
                ]
            ]
        }
    ]]`
    
    when I am converting it says ✔ Generated output file (0 documents): spacy_training_data.spacy

**Reply by rmitsch** [reply]
Hi @sunilksamanta, please open a new discussion and provide more information on your use case. Thanks!
